src/CommunityToolkit.Maui/HandlerImplementation/GravatarImageSourceService/GravatarImageSourceService.android.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Android.Content;
+using Android.Graphics.Drawables;
+using CommunityToolkit.Maui.Core;
+using Microsoft.Extensions.Logging;
+
+namespace CommunityToolkit.Maui.Views;
+
+public partial class GravatarImageSourceService
+{
+	public override async Task<IImageSourceServiceResult?> LoadDrawableAsync(IImageSource imageSource, Android.Widget.ImageView imageView, CancellationToken cancellationToken = default)
+	{
+		IGravatarImageSource gravatarImageSource = (IGravatarImageSource)imageSource;
+		if (!gravatarImageSource.IsEmpty && gravatarImageSource.Uri is not null)
+		{
+			try
+			{
+				ImageLoaderCallback callback = new();
+				PlatformInterop.LoadImageFromUri(imageView, gravatarImageSource.Uri.OriginalString, new Java.Lang.Boolean(gravatarImageSource.CachingEnabled), callback);
+				return await callback.Result;
+			}
+			catch (Exception ex)
+			{
+				Logger?.LogWarning(ex, "Unable to load image URI '{Uri}'.", gravatarImageSource.Uri.OriginalString);
+				throw;
```

Comment:
Can we use some default image? The app crashes if there is no internet connection.

Reply:
I have changed this code to link\use the `UriImageSourceService`.